src/coreclr/jit/forwardsub.cpp

```diff
@@ -495,24 +481,12 @@ bool Compiler::fgForwardSubStatement(Statement* stmt)
         return false;
     }
 
-    // Only fwd sub if we expect no code duplication
+    // Cannot forward sub without liveness information.
     //
-    bool livenessBased = false;
-    if (varDsc->lvRefCnt(RCS_EARLY) != 2)
+    if (!fgDidEarlyLiveness)
```

Comment:
Bail up in `fgForwardSub`, and assert `fgDidEarlyLiveness' is true here?

Reply:
Makes sense, done